scripts/install/install_unix.sh

```diff
@@ -13,165 +13,638 @@
+    out_file_path="$TMP_DIR/$package_file_name"
+}
+
+set_proxy()
+{
+  if [ -n "$proxy" ]; then
+    info "Using proxy from arguments: $proxy"
+    if [ -n "$proxy_user" ]; then
+      while [ -z "$proxy_password" ] && [ ! "$accept_defaults" = "yes" ]; do
+        increase_indent
+        command printf "${indent}$(fg_blue "$proxy_user@$proxy")'s password: "
+        stty -echo
+        read -r proxy_password
+        stty echo
+        info
+        if [ -z "$proxy_password" ]; then
+          warn "The password must be provided!"
+        fi
+        decrease_indent
+      done
+      protocol="$(echo "$proxy" | cut -d'/' -f1)"
+      host="$(echo "$proxy" | cut -d'/' -f3)"
+      full_proxy="$protocol//$proxy_user:$proxy_password@$host"
+    fi
+
+  elif [ -n "$http_proxy" ]; then
+    info "Using proxy from profile: $http_proxy"
+    proxy="$http_proxy"
+  elif [ -n "$https_proxy" ]; then
+    info "Using proxy from profile: $https_proxy"
+    proxy="$https_proxy"
+  fi
+
+  if [ -z "$full_proxy" ]; then
+    full_proxy="$proxy"
+  fi
+}
+
+# This will set the os based on the current runtime environment.
+# Accepted values are darwin and linux. This value cannot be overriden.
+set_os()
+{
+  os_key=$(uname -s)
+  case "$os_key" in
+    Linux)
+      os="linux"
+      ;;
+    *)
+      error "Unsupported os type: $os_key"
+      ;;
+  esac
+}
 
-set_arch() {
-    os_arch=$(uname -m)
-    case "$os_arch" in
+
+set_os_arch()
+{
+  os_arch=$(uname -m)
+  case "$os_arch" in 
     # arm64 strings. These are from https://stackoverflow.com/questions/45125516/possible-values-for-uname-m
     aarch64|arm64|aarch64_be|armv8b|armv8l)
-        arch="arm64"
-        ;;
+      os_arch="arm64"
+      ;;
     x86_64)
-        arch="amd64"
-        ;;
+      os_arch="amd64"
+      ;;
     # armv6/32bit. These are what raspberry pi can return, which is the main reason we support 32-bit arm
     arm|armv6l|armv7l)
-        arch="arm"
-        ;;
+      os_arch="arm"
+      ;;
     *)
-        echo "Installation failed: Unsupported os arch: $os_arch"
-        exit 1
-        ;;
-    esac
+      error "Unsupported os arch: $os_arch"
+      ;;
+  esac
+}
+
+# Set the package type before install
+set_package_type()
+{
+  if command -v dpkg > /dev/null 2>&1; then
+      package_type="deb"
+  elif command -v rpm > /dev/null 2>&1; then
+      package_type="rpm"
+  else
+      error "Could not find dpkg or rpm on the system"
+  fi
+}
+
+# This will set the urls to use when downloading the collector and its plugins.
+# These urls are constructed based on the --version flag or COLLECTOR_VERSION env variable.
+# If not specified, the version defaults to "latest".
+set_download_urls()
+{
+  if [ -z "$version" ] ; then
+    # shellcheck disable=SC2153
+    version=$COLLECTOR_VERSION
+  fi
+
+  if [ -z "$url" ] ; then
+    url=$DOWNLOAD_BASE
+  fi
+
+  if [ -z "$arch" ] ; then 
+    os_arch=$arch
+  fi
+
+  if [ -z "$pkg_type" ] ; then
+    package_type=$pkg_type
+  fi
```

Comment:
These don't make sense to me; If "$arch" is empty, "$os_arch" would be set to empty? I think these two if blocks probably shouldn't exist?